packages/gatsby-cli/src/create-cli.ts

```diff
@@ -502,7 +503,15 @@ export const createCli = (argv: Array<string>): yargs.Arguments => {
         const starterStr = starter ? String(starter) : undefined
         const rootPathStr = rootPath ? String(rootPath) : undefined
 
-        await initStarter(starterStr, rootPathStr)
+        if (
+          process.env.GATSBY_EXPERIMENTAL_GATSBY_NEW_FLOW &&
+          !starterStr &&
+          !rootPathStr
+        ) {
```

Comment:
Nitpicky again:

Would you mind adding an explicit comment on why this condition is here? 

`We only run the interactive CLI when there are no arguments in gatsby new`

Or storing the result of the condition in an intermediate variable that is self explanatory? For clarity reasons. I know test is covering these scenario, but personally, I would love to read the code and get what's going on directly :D